File: src/main/java/com/milaboratory/mixcr/trees/IdGenerator.java
package com.milaboratory.mixcr.trees;

import java.util.concurrent.atomic.AtomicInteger;

class IdGenerator {
    private final AtomicInteger counter = new AtomicInteger(0);

    public TreeWithMetaBuilder.TreeId next(VJBase VJBase) {
        return new TreeWithMetaBuilder.TreeId(counter.incrementAndGet(), VJBase);
    }
}

File: src/main/kotlin/com/milaboratory/mixcr/trees/IdGenerator.kt
package com.milaboratory.mixcr.trees

import com.milaboratory.mixcr.trees.TreeWithMetaBuilder.TreeId
import java.util.concurrent.atomic.AtomicInteger

internal class IdGenerator {
    private val counter = AtomicInteger(0)
    fun next(VJBase: VJBase): TreeId = TreeId(counter.incrementAndGet(), VJBase)
}
